### PR TITLE
Hiding Task as internal implementation

### PR DESCRIPTION
**Pull request by tjaskula**
This has the following goals

- Remove `Task` as return type from interfaces and public methods
- Implement fire-and-forget async operations which allows to not propagate the task to the up. If `Task` throws it will be matched and logged.
- Only `SocketChannelWriter` was made synchronous because it is impossible to do otherwise.

@VaughnVernon @zpbappi Please review if you have any suggestions or comments. Please don't merge it for now as I need to do more testing.

### CODE REVIEW

File: src/Vlingo.Wire/Vlingo.Wire.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <!-- NuGet Metadata -->
     <IsPackable>true</IsPackable>
-    <PackageVersion>0.4.2</PackageVersion>
+    <PackageVersion>0.4.3</PackageVersion>
```
Comment (tjaskula) [inline src/Vlingo.Wire/Vlingo.Wire.csproj]: It's a breaking change I need to bump it to `0.5`

File: src/Vlingo.Wire.Tests/Fdx/Bidirectional/SocketRequestResponseChannelTest.cs [hunk 2]
```diff
@@ -155,11 +157,13 @@ public async Task TestThatRequestResponsePoolLimitsNotExceeded()
...
             for (int idx = 0; idx < total; ++idx) {
-                await Request(request + idx.ToString("D3"));
+                Request(request + idx.ToString("D3"));
+                Thread.Sleep(10);
             }
     
             while (_clientConsumer.UntilConsume.Remaining > 0) {
-                await _client.ProbeChannel();
+                _client.ProbeChannel();
+                Thread.Sleep(30);
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Bidirectional/SocketRequestResponseChannelTest.cs]: I had to add these in several places as now it's async fire and forget so the tests finishes before it get a chance to handle the message :)

**Comment by tjaskula** [comment]
I need to probably address the CI UT issues, as there are many failing tests (but not locally).

**Comment by tjaskula** [comment]
@VaughnVernon I'll have a look on the code. I'm not sure I can use Actors on that level as it at socket level but certainly I'll give it a try to convert it to something better. I'll keep you posted

**Comment by VaughnVernon** [comment]
> @VaughnVernon I'll have a look on the code. I'm not sure I can use Actors on that level as it at socket level but certainly I'll give it a try to convert it to something better. I'll keep you posted

@tjaskula Just so you understand, I am suggesting taking the socket probing/reading and writing *out* of actors. You could use AsyncMessageQueue, which is async but won't cause a hold up on actor mailboxes. It's a different technique all together. 

- When you read input from the socket you send the data to the actor (the actor does not socket interactions)
- If there are any exceptions while reading and/or writing, you can send the exception to the actor so the actor can throw the exception for supervision. Then the supervision can do something to help the socket reader/writer via AsyncMessageQueue.

**Comment by tjaskula** [comment]
@VaughnVernon So everything is asynchronous and has been moved to APM pattern and it's green 🍏 . Please review and change your status if it's ok. **Don't merge it yet I'll do it by myself**.

@zpbappi if you have any comments or improvements I'll be happy to read them :)

**Pull request merged**